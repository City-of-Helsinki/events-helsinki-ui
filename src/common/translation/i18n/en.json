{
  "about": {
    "title": "About the service"
  },
  "accessibility": {
    "title": "Accessibility Statement"
  },
  "appName": "Events",
  "commons": {
    "autosuggest": {
      "menu": {
        "ariaButtonClose": "Close menu",
        "info": "You can also search by the keyword you entered",
        "title": "Search suggestions"
      }
    },
    "category": {
      "ariaButtonRemove": "Remove category: {{category}}"
    },
    "dateSelector": {
      "dateTypeAll": "At any day",
      "dateTypeThisWeek": "This week",
      "dateTypeToday": "Today",
      "dateTypeTomorrow": "Tomorrow",
      "dateTypeWeekend": "This weekend",
      "infoDate": "dd.mm.yyyy",
      "labelEndDate": "End date",
      "labelStartDate": "Start date",
      "menu": {
        "buttonBack": "Back",
        "buttonClose": "Close",
        "buttonCustom": "Select dates"
      },
      "title": "Select dates"
    },
    "dropdown": {
      "menu": {
        "buttonClear": "Clear selection"
      }
    },
    "filter": {
      "ariaButtonRemove": "Remove filter: {{filter}}"
    },
    "linkSkipToContent": "Skip to content",
    "loading": "Loading...",
    "multiSelectDropdown": {
      "inputPlaceholder": "Write search word",
      "selectAll": "Select all"
    },
    "preview": "Preview",
    "shareLink": {
      "shareOnFacebook": "Share on Facebook",
      "shareOnTwitter": "Share on Twitter",
      "shareOnLinkedIn": "Share on LinkedIn"
    },
    "shareLinks": {
      "buttonCopyLink": "Copy link address",
      "messageLinkCopySuccess": "Link copied into clipboard"
    },
    "supriseMe": "I'm Feeling Lucky",
    "timeAbbreviation": "at"
  },
  "collection": {
    "buttonShowAllPastEvents": "Show all past recommendations",
    "collectionCard": {
      "ariaLabelLink": "Go to collection: {{title}}",
      "count": "{{count}} hits"
    },
    "curatedEvents": {
      "linkOnlyExpiredEvents": "To all collections",
      "textEventNotFound": "Event not found",
      "textOnlyExpiredEvents": "Unfortunately all the events in this collection have already passed. Have a look at our other recommendations and collections to find more interesting events."
    },
    "eventList": {
      "errorLoadMore": "Failed to load events"
    },
    "expired": {
      "linkSearchEvents": "To all collections",
      "text": "Unfortunately this particular collections has already been removed. Luckily there are many other collections and recommendations, so have a look and explore all the other interesting things the city has to offer.",
      "title": "This collection has been removed"
    },
    "languageNotSupported": {
      "linkSearchEvents": "Go to all collections",
      "text": "This content is not available in the language you are looking for. Luckily there are many other recommendations, so have a look at the other collections to find loads of interesting events.",
      "title": "This collection is not available in English"
    },
    "notFound": {
      "linkSearchEvents": "Find something to do",
      "text": "For some reason we can't find the collection you are looking for. Try searching for something else here:",
      "title": "Oops, you got hit by a 404!"
    },
    "shareLinks": {
      "title": "Share collection"
    },
    "title": "Loading collection...",
    "titlePastRecommendations": "Past recommendations",
    "titleSimilarCollections": "More recommendations for you"
  },
  "collectionList": {
    "pageTitle": "See our recommendations",
    "title": "Our suggestions for you"
  },
  "event": {
    "categories": {
      "labelFree": "Free",
      "labelThisWeek": "This week",
      "labelToday": "Today"
    },
    "description": {
      "title": "Description"
    },
    "eventCancelled": "Event cancelled",
    "eventCard": {
      "ariaLabelBuyTickets": "Buy tickets - The link will open in a new window",
      "ariaLabelLink": "Go to event: {{name}}",
      "ariaLabelReadMore": "Read more: {{name}}",
      "buttonBuyTickets": "Buy tickets",
      "buttonReadMore": "Read more",
      "dateAndTime": "{{date}}, at {{time}}",
      "isFree": "Free"
    },
    "hero": {
      "ariaLabelBackButton": "Go back to previous page",
      "ariaLabelBuyTickets": "Buy tickets - The link will open in a new window",
      "buttonBuyTickets": "Buy tickets",
      "buttonToHomePage": "Visit the landing page",
      "offers": {
        "isFree": "Free"
      },
      "textEventClosed": "Worry not, there are lots of ther things to do in Helsinki.",
      "titleEventClosed": "Oh no, the event you are looking for has already ended"
    },
    "info": {
      "ariaLabelBuyTickets": "Buy tickets - The link will open in a new window",
      "buttonAddToCalendar": "Add to calendar",
      "buttonBuyTickets": "Buy tickets",
      "directionsGoogle": "Directions (Google)",
      "directionsHSL": "Directions (HSL)",
      "errorLoadMore": "Failed to load events",
      "extlinkFacebook": "Facebook",
      "extlinkInstagram": "Instagram",
      "extlinkTwitter": "Twitter",
      "extlinkYoutube": "Youtube",
      "labelDateAndTime": "Date and time",
      "labelDirections": "Find your way",
      "labelLanguages": "Languages",
      "labelLocation": "Location",
      "labelOrganizer": "Organizer information",
      "labelOtherInfo": "Other information",
      "labelPrice": "Price",
      "linkSearchByOrganization": "See other events by organizer",
      "linkWebPage": "Web page",
      "offers": {
        "isFree": "Free"
      },
      "openMap": "Open the map",
      "textCalendarLinkDescription": "{{description}} Read more: {{link}}",
      "title": "Loading event..."
    },
    "location": {
      "directionsGoogle": "Directions (Google)",
      "directionsHSL": "Directions (HSL)",
      "mapTitle": "Location",
      "openMap": "Open the map in a new window",
      "title": "Location"
    },
    "notFound": {
      "linkSearchEvents": "Find something to do",
      "text": "For some reason we can't find the event you are looking for. Try searching for something else here:",
      "title": "Oops, you got hit by a 404!"
    },
    "otherTimes": {
      "buttonHide": "Hide other event times",
      "buttonReadMore": "Read more; {{date}}",
      "buttonShow": "Show other event times",
      "title": "Other times of the event"
    },
    "shareLinks": {
      "title": "Share Event"
    },
    "similarEvents": {
      "title": "Similar events"
    }
  },
  "eventSearch": {
    "ariaLiveLoading": "Searching for events",
    "ariaLiveSearchReady": "{{count}} search results",
    "buttonClearFilters": "Clear filters",
    "buttonLoadMore": "Show more events ({{count}})",
    "errorLoadMore": "Failed to load events",
    "noResultsInfo": {
      "bigText": "Sorry, no search results were found for the search criteria you selected",
      "smallText": "Try changing your search criteria and you're sure to find something to do!"
    },
    "search": {
      "buttonSearch": "Search",
      "checkboxIsFree": "Show only events that are free",
      "checkboxOnlyChildrenEvents": "Show only events for children",
      "checkboxOnlyEveningEvents": "Show only evening events",
      "labelCategory": "What kind of event you are looking for?",
      "labelDateRange": "For what period you are looking for?",
      "labelDistrict": "From where are you looking for?",
      "labelSearchField": "What are you looking for",
      "labelTargetGroup": "Whom you are looking for?",
      "placeholder": "Enter a keyword, eg rock or yoga",
      "selectAllDivisions": "All areas",
      "selectAllPlaces": "All venues",
      "titleDropdownCategory": "Select category",
      "titleDropdownDivision": "Search area",
      "titleDropdownPlace": "Search venue"
    },
    "textFoundEvents": "{{count}} search results",
    "title": "Event search"
  },
  "footer": {
    "linkAbout": "About the service",
    "linkAccessibility": "Accessibility Statement",
    "linkFeedback": "Give feedback",
    "linkFeedbackUrl": "https://www.hel.fi/helsinki/en/administration/participate/feedback/",
    "searchCollections": "We recommend",
    "searchEvents": "Find something to do",
    "titleCategories": "Categories"
  },
  "header": {
    "ariaButtonCloseMenu": "Close menu",
    "ariaButtonOpenMenu": "Open menu",
    "ariaLabelLogo": "Home",
    "changeLanguage": "English - language selection",
    "languages": {
      "en": "In English",
      "fi": "Suomeksi",
      "sv": "PÃ¥ svenska"
    },
    "searchCollections": "We recommend",
    "searchEvents": "Find something to do"
  },
  "home": {
    "category": {
      "culture": "Arts and culture",
      "dance": "Dance",
      "food": "Food",
      "influence": "Participate and influence",
      "misc": "Other",
      "movie": "Movies",
      "museum": "Museums",
      "music": "Music",
      "nature": "Nature and outdoor activities",
      "sport": "Exercise and sports",
      "theatre": "Theatre"
    },
    "collections": {
      "title": "See our recommendations"
    },
    "notFound": {
      "text": "For some reason we can't find the landing page you are looking for.",
      "title": "Oops, you got hit by a 404!"
    },
    "search": {
      "buttonSearch": "Search",
      "labelDateRange": "For what period you are looking for?",
      "labelSearchField": "What are you looking for",
      "linkAdvancedSearch": "Advanced search",
      "placeholder": "Enter a keyword, eg rock or yoga",
      "title": "Find something to do"
    }
  },
  "menu": {
    "aboutService": "About the service",
    "mostPopular": "Top searches",
    "searchCategories": "Browse Categories",
    "searchEvents": "Find something to do"
  },
  "notFound": {
    "linkSearchEvents": "Find something to do",
    "text": "For some reason we can't find the page you are looking for. Try searching for something else here:",
    "title": "Oops, you got hit by a 404!"
  }
}
