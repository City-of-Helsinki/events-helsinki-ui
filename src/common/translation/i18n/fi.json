{
  "about": {
    "title": "Tietoa palvelusta"
  },
  "accessibility": {
    "title": "Saavutettavuus­seloste"
  },
  "appName": "Tapahtumat",
  "commons": {
    "autosuggest": {
      "menu": {
        "ariaButtonClose": "Sulje valikko",
        "info": "Voit myös etsiä kirjoittamallasi hakusanalla",
        "title": "Hakuehdotuksia"
      }
    },
    "category": {
      "ariaButtonRemove": "Poista kategoria: {{category}}"
    },
    "dateSelector": {
      "dateTypeAll": "Koska tahansa",
      "dateTypeThisWeek": "Tällä viikolla",
      "dateTypeToday": "Tänään",
      "dateTypeTomorrow": "Huomenna",
      "dateTypeWeekend": "Viikonloppuna",
      "infoDate": "pp.kk.vvvv",
      "labelEndDate": "Loppumispäivä",
      "labelStartDate": "Alkamispäivä",
      "menu": {
        "buttonBack": "Takaisin",
        "buttonClose": "Sulje",
        "buttonCustom": "Valitse päivät"
      },
      "title": "Valitse ajankohta"
    },
    "dropdown": {
      "menu": {
        "buttonClear": "Tyhjennä valinnat"
      }
    },
    "filter": {
      "ariaButtonRemove": "Poista suodatin: {{filter}}"
    },
    "linkSkipToContent": "Siirry sisältöön",
    "loading": "Ladataan...",
    "multiSelectDropdown": {
      "inputPlaceholder": "Kirjoita hakusana",
      "selectAll": "Valitse kaikki"
    },
    "preview": "Esikatselu",
    "shareLink": {
      "shareOnFacebook": "Jaa Facebookissa",
      "shareOnTwitter": "Jaa Twitterissä",
      "shareOnLinkedIn": "Jaa LinkedInissä"
    },
    "shareLinks": {
      "buttonCopyLink": "Kopioi linkin osoite",
      "messageLinkCopySuccess": "Linkki kopioitu leikepöydälle"
    },
    "supriseMe": "Koeta onneasi",
    "timeAbbreviation": "klo"
  },
  "collection": {
    "buttonShowAllPastEvents": "Näytä kaikki menneet",
    "collectionCard": {
      "ariaLabelLink": "Siirry kokoelmaan: {{title}}",
      "count": "{{count}} kpl"
    },
    "curatedEvents": {
      "linkOnlyExpiredEvents": "Tutustu kaikkiin kokoelmiin",
      "textEventNotFound": "Tapahtumaa ei löydy",
      "textOnlyExpiredEvents": "Valitettavasti tämän kokoelman kaikki suositellut tapahtumat ovat jo ehtineet päättyä. Käy tutustumassa muihin suosituksiimme ja löydä itsellesi tekemistä."
    },
    "eventList": {
      "errorLoadMore": "Tapahtumien lataaminen epäonnistui"
    },
    "expired": {
      "linkSearchEvents": "Tutustu kaikkiin kokoelmiin",
      "text": "Valitettavasti etsimäsi kokoelma on poistettu. Onneksi uusia kokoelmia syntyy koko ajan, joten käy kurkkaamassa mitä muuta kiinnostavaa kaupungissa tapahtuu.",
      "title": "Etsimäsi kokoelma on poistettu"
    },
    "languageNotSupported": {
      "linkSearchEvents": "Tutustu kaikkiin kokoelmiin",
      "text": "Valitettavasti etsimääsi kokoelmaa ei ole saatavilla kaikilla kielillä. Onneksi tekeminen ei lopu tähän, käy selaamassa muita kokoelmia ja löydä kiinnostavia tapahtumia.",
      "title": "Kokoelmaa ei löydy suomeksi"
    },
    "notFound": {
      "linkSearchEvents": "Hae itsellesi tekemistä",
      "text": "Syystä tai toisesta hakemaasi kokoelmaa ei löydy. Löydä itsellesi tekemistä vaikkapa täältä:",
      "title": "Oho, nyt iski 404!"
    },
    "shareLinks": {
      "title": "Jaa kokoelma"
    },
    "title": "Ladataan kokoelmaa...",
    "titlePastRecommendations": "Menneet suositukset",
    "titleSimilarCollections": "Suosittelemme myös näitä"
  },
  "collectionList": {
    "pageTitle": "Suosittelemme",
    "title": "Kokosimme sinulle tekemistä"
  },
  "event": {
    "categories": {
      "labelFree": "Maksuton",
      "labelThisWeek": "Tällä viikolla",
      "labelToday": "Tänään"
    },
    "description": {
      "title": "Kuvaus"
    },
    "eventCancelled": "Tapahtuma peruttu",
    "eventCard": {
      "ariaLabelBuyTickets": "Osta liput - Linkki avautuu uudessa ikkunassa",
      "ariaLabelLink": "Siirry tapahtumaan: {{name}}",
      "ariaLabelReadMore": "Lue lisää: {{name}}",
      "buttonBuyTickets": "Osta liput",
      "buttonReadMore": "Lue lisää",
      "dateAndTime": "{{date}}, klo {{time}}",
      "isFree": "Maksuton"
    },
    "hero": {
      "ariaLabelBackButton": "Palaa takaisin edelliselle sivulle",
      "ariaLabelBuyTickets": "Osta liput - Linkki avautuu uudessa ikkunassa",
      "buttonBuyTickets": "Osta liput",
      "buttonToHomePage": "Siirry etusivulle",
      "offers": {
        "isFree": "Maksuton"
      },
      "textEventClosed": "Älä huoli, Helsingistä löytyy paljon muuta tekemistä.",
      "titleEventClosed": "Höh, etsimäsi tapahtuma on jo ehtinyt päättyä"
    },
    "info": {
      "ariaLabelBuyTickets": "Osta liput - Linkki avautuu uudessa ikkunassa",
      "buttonAddToCalendar": "Lisää kalenteriin",
      "buttonBuyTickets": "Osta liput",
      "directionsGoogle": "Reittiohjeet (Google)",
      "directionsHSL": "Reittiohjeet (HSL)",
      "errorLoadMode": "Tapahtumien lataaminen epäonnistui",
      "extlinkFacebook": "Facebook",
      "extlinkInstagram": "Instagram",
      "extlinkTwitter": "Twitter",
      "extlinkYoutube": "Youtube",
      "labelDateAndTime": "Päivä ja aika",
      "labelDirections": "Löydä perille",
      "labelLanguages": "Kielet",
      "labelLocation": "Paikka",
      "labelOrganizer": "Järjestäjän tiedot",
      "labelOtherInfo": "Muut tiedot",
      "labelPrice": "Hinta",
      "linkSearchByOrganization": "Katso muut järjestäjän tapahtumat",
      "linkWebPage": "Verkkosivu",
      "offers": {
        "isFree": "Maksuton"
      },
      "openMap": "Avaa kartta",
      "textCalendarLinkDescription": "{{description}} Lue lisää: {{link}}"
    },
    "location": {
      "directionsGoogle": "Reittiohjeet (Google)",
      "directionsHSL": "Reittiohjeet (HSL)",
      "mapTitle": "Sijainti",
      "openMap": "Avaa kartta uuteen ikkunaan",
      "title": "Sijainti"
    },
    "notFound": {
      "linkSearchEvents": "Hae itsellesi tekemistä",
      "text": "Syystä tai toisesta hakemaasi tapahtumaa ei löydy. Löydä itsellesi tekemistä vaikkapa täältä:",
      "title": "Oho, nyt iski 404!"
    },
    "otherTimes": {
      "buttonHide": "Piilota muut tapahtuma-ajat",
      "buttonReadMore": "Lue lisää: {{date}}",
      "buttonShow": "Näytä muut tapahtuma-ajat",
      "title": "Tapahtuman muut ajat"
    },
    "shareLinks": {
      "title": "Jaa tapahtuma"
    },
    "similarEvents": {
      "title": "Muita samankaltaisia tapahtumia"
    },
    "title": "Ladataan tapahtumaa..."
  },
  "eventSearch": {
    "ariaLiveLoading": "Haetaan tapahtumia",
    "ariaLiveSearchReady": "{{count}} hakutulosta",
    "buttonClearFilters": "Tyhjennä hakuehdot",
    "buttonLoadMore": "Näytä lisää tapahtumia ({{count}})",
    "errorLoadMode": "Tapahtumien lataaminen epäonnistui",
    "noResultsInfo": {
      "bigText": "Voi vitsi, valitsemillasi hakuehdoilla ei löytynyt yhtään hakutulosta",
      "smallText": "Kokeile muuttamalla hakuehtoja ja varmasti löydät jotain kiinnostavaa tekemistä!"
    },
    "search": {
      "buttonSearch": "Hae",
      "checkboxIsFree": "Näytä vain maksuttomat",
      "checkboxOnlyChildrenEvents": "Näytä vain lastentapahtumat",
      "checkboxOnlyEveningEvents": "Näytä vain iltatapahtumat",
      "labelCategory": "Millaista haet?",
      "labelDateRange": "Mille ajankohdalle etsit?",
      "labelDistrict": "Mistä etsit?",
      "labelSearchField": "Mitä etsit?",
      "labelTargetGroup": "Kenelle etsit?",
      "placeholder": "Kirjoita hakusana, esim. rock tai jooga",
      "selectAllDivisions": "Kaikki alueet",
      "selectAllPlaces": "Kaikki tapahtumapaikat",
      "titleDropdownCategory": "Valitse kategoria",
      "titleDropdownDivision": "Etsi alue",
      "titleDropdownPlace": "Etsi tapahtumapaikka"
    },
    "textFoundEvents": "{{count}} hakutulosta",
    "title": "Tapahtumahaku"
  },
  "footer": {
    "linkAbout": "Tietoa palvelusta",
    "linkAccessibility": "Saavutettavuus­seloste",
    "linkFeedback": "Anna palautetta",
    "linkFeedbackUrl": "https://www.hel.fi/helsinki/fi/kaupunki-ja-hallinto/osallistu-ja-vaikuta/palaute/anna-palautetta",
    "searchCollections": "Suosittelemme",
    "searchEvents": "Etsi tekemistä",
    "titleCategories": "Kategoriat"
  },
  "header": {
    "ariaButtonCloseMenu": "Sulje valikko",
    "ariaButtonOpenMenu": "Avaa valikko",
    "ariaLabelLogo": "Etusivu",
    "changeLanguage": "Suomi - kielivalikko",
    "languages": {
      "en": "In English",
      "fi": "Suomeksi",
      "sv": "På svenska"
    },
    "searchCollections": "Suosittelemme",
    "searchEvents": "Etsi tekemistä"
  },
  "home": {
    "category": {
      "culture": "Kulttuuri ja taide",
      "dance": "Tanssi",
      "food": "Ruoka",
      "influence": "Osallistu ja vaikuta",
      "misc": "Muut",
      "movie": "Elokuva",
      "museum": "Museot",
      "music": "Musiikki",
      "nature": "Luonto ja ulkoilu",
      "sport": "Liikunta ja urheilu",
      "theatre": "Teatteri"
    },
    "collections": {
      "title": "Suosittelemme"
    },
    "notFound": {
      "text": "Syystä tai toisesta hakemaasi kotisivua ei löydy.",
      "title": "Oho, nyt iski 404!"
    },
    "search": {
      "buttonSearch": "Hae",
      "labelDateRange": "Mille ajankohdalle etsit?",
      "labelSearchField": "Mitä etsit?",
      "linkAdvancedSearch": "Tarkennettu haku",
      "placeholder": "Kirjoita hakusana, esim. rock tai jooga",
      "title": "Löydä tekemistä"
    }
  },
  "menu": {
    "aboutService": "Tietoa palvelusta",
    "mostPopular": "Suosituimmat haut",
    "searchCategories": "Selaa kategorioita",
    "searchEvents": "Etsi tekemistä"
  },
  "notFound": {
    "linkSearchEvents": "Hae itsellesi tekemistä",
    "text": "Syystä tai toisesta hakemaasi sivua ei löydy. Löydä itsellesi tekemistä vaikkapa täältä:",
    "title": "Oho, nyt iski 404!"
  }
}
