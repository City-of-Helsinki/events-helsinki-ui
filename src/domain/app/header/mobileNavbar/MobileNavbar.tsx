import classNames from 'classnames';
import { IconCross, IconMenuHamburger } from 'hds-react';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

import useLocale from '../../../../hooks/useLocale';
import { ROUTES } from '../../constants';
import Container from '../../layout/Container';
import styles from './mobileNavbar.module.scss';

interface Props {
  isMenuOpen: boolean;
  onCloseMenu: () => void;
  onToggleMenu: () => void;
}

const MobileNavbar: React.FC<Props> = ({
  isMenuOpen,
  onCloseMenu,
  onToggleMenu,
}) => {
  const { t } = useTranslation();
  const locale = useLocale();

  const logoLang = locale === 'sv' ? 'sv' : 'fi';

  return (
    <Container>
      <div className={styles.mobileNavbar}>
        <Link
          aria-label={t('header.ariaLabelLogo')}
          onClick={onCloseMenu}
          to={`/${locale}${ROUTES.HOME}`}
          className={styles.logoWrapper}
        >
          <div className={classNames(styles.logo, styles[logoLang])} />
          <div className={styles.appName}>{t('appName')}</div>
        </Link>
        <div className={styles.buttonWrapper}>
          <button
            className={styles.menuButton}
            onClick={onToggleMenu}
            aria-label={
              isMenuOpen
                ? t('header.ariaButtonCloseMenu')
                : t('header.ariaButtonOpenMenu')
            }
          >
            {isMenuOpen ? <IconCross aria-hidden /> : <IconMenuHamburger aria-hidden />}
          </button>
        </div>
      </div>
    </Container>
  );
};

export default MobileNavbar;
