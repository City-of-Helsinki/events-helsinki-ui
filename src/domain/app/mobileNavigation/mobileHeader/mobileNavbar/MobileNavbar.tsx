import { IconClose, IconMenu } from "hds-react";
import * as React from "react";
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router-dom";

import styles from "./mobileNavbar.module.scss";

interface Props {
  isMenuOpen: boolean;
  onCloseMenu: () => void;
  onOpenMenu: () => void;
}

const MobileNavbar: React.FC<Props> = ({
  isMenuOpen,
  onCloseMenu,
  onOpenMenu
}) => {
  const { t } = useTranslation();
  const history = useHistory();
  const moveHome = () => {
    history.push("/");
    onCloseMenu();
  };

  return (
    <div className={styles.mobileNavbar}>
      <div className={styles.logoWrapper}>
        <div className={styles.logo} onClick={moveHome}></div>
        <h1 className={styles.appName}>{t("appName")}</h1>
      </div>
      <div className={styles.buttonWrapper}>
        {isMenuOpen ? (
          <>
            <button
              className={styles.closeButton}
              onClick={onCloseMenu}
              aria-label={t("header.ariaButtonCloseMenu")}
            >
              <IconClose />
            </button>
          </>
        ) : (
          <button
            className={styles.menuButton}
            onClick={onOpenMenu}
            aria-label={t("header.ariaButtonOpenMenu")}
          >
            <IconMenu />
          </button>
        )}
      </div>
    </div>
  );
};

export default MobileNavbar;
