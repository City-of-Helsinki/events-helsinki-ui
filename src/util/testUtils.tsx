import { fireEvent } from '@testing-library/react';

export const arrowUpKeyPressHelper = () =>
  fireEvent.keyDown(document, { code: 38, key: 'ArrowUp' });

export const arrowDownKeyPressHelper = () =>
  fireEvent.keyDown(document, { code: 40, key: 'ArrowDown' });

export const escKeyPressHelper = () =>
  fireEvent.keyDown(document, { code: 27, key: 'Escape' });

export const tabKeyPressHelper = () =>
  fireEvent.keyDown(document, { code: 9, key: 'Tab' });
