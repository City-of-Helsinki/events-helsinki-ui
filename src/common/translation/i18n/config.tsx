import { InitOptions } from "i18next";

import en from "./en.json";
import fi from "./fi.json";
import sv from "./sv.json";

const i18Config: InitOptions = {
  fallbackLng: "fi",
  interpolation: {
    escapeValue: false
  },
  load: "languageOnly",
  preload: ["fi", "en", "sv"],
  react: {
    useSuspense: false
  },
  resources: {
    en: {
      translation: en
    },
    fi: {
      translation: fi
    },
    sv: {
      translation: sv
    }
  },
  saveMissing: true
};

export default i18Config;
