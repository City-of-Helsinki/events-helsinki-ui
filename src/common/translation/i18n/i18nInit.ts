import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import en from "./en.json";
import fi from "./fi.json";
import sv from "./sv.json";

i18n.use(initReactI18next).init({
  fallbackLng: "fi",
  interpolation: {
    escapeValue: false
  },
  lng: "fi",
  resources: {
    en: {
      translation: en
    },
    fi: {
      translation: fi
    },
    sv: {
      translation: sv
    }
  }
});

export default i18n;
