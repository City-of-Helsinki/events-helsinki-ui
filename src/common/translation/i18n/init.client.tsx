import i18next from "i18next";
import BrowserLanguageDetector from "i18next-browser-languagedetector";
import { initReactI18next } from "react-i18next";

import config from "./config";

const languageDetectorOptions = {
  caches: ["cookie"],
  // order and from where user language should be detected
  order: ["path", "cookie", "navigator"]
};

if (!i18next.isInitialized) {
  i18next
    .use(initReactI18next)
    .use(BrowserLanguageDetector)
    .init({ ...config, detection: languageDetectorOptions });
}

export default i18next;
