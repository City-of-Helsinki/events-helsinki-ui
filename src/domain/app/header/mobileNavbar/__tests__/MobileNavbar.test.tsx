import i18n from 'i18next';
import { axe } from 'jest-axe';
import React from 'react';

import translations from '../../../../../common/translation/i18n/fi.json';
import { render, screen } from '../../../../../util/testUtils';
import MobileNavbar from '../MobileNavbar';

beforeEach(() => {
  i18n.changeLanguage('fi');
});

const getWrapper = () =>
  render(<MobileNavbar isMenuOpen={true} onToggleMenu={jest.fn()} />);

it('component should be accessible', async () => {
  const { container } = getWrapper();
  expect(await axe(container)).toHaveNoViolations();
});

it('should show Swedish icon', () => {
  i18n.changeLanguage('sv');
  const { container } = render(
    <MobileNavbar isMenuOpen={true} onToggleMenu={jest.fn()} />
  );

  expect(container.firstChild).toMatchSnapshot();
  expect(
    screen.queryByRole('link', { name: translations.header.ariaLabelLogo })
      .firstChild
  ).toHaveClass('sv');
});
