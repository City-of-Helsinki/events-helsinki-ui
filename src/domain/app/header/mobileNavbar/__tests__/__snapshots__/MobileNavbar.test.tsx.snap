// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show Swedish icon 1`] = `
<div
  class="container"
>
  <div
    class="mobileNavbar"
  >
    <a
      aria-label="Etusivu"
      class="logoWrapper"
      href="/sv/home"
    >
      <div
        class="logo sv"
      />
      <div
        class="appName"
      >
        Tapahtumat
      </div>
    </a>
    <div
      class="buttonWrapper"
    >
      <button
        aria-label="Sulje valikko"
        class="menuButton"
      >
        <svg
          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <rect
              height="24"
              width="24"
            />
            <polygon
              fill="currentColor"
              points="18 7.5 13.5 12 18 16.5 16.5 18 12 13.5 7.5 18 6 16.5 10.5 12 6 7.5 7.5 6 12 10.5 16.5 6"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
</div>
`;
